spec/features/school_student/deleting_a_school_student_spec.rb

```diff
@@ -39,6 +50,22 @@
     expect(response).to have_http_status(:forbidden)
   end
 
+  it 'does not create the school owner safeguarding flag when logged in as a teacher' do
+    teacher = create(:teacher, school:)
+    authenticated_in_hydra_as(teacher)
+
+    delete("/api/schools/#{school.id}/students/#{student_id}", headers:)
+    expect(ProfileApiClient).not_to have_received(:create_safeguarding_flag).with(token: UserProfileMock::TOKEN, flag: ProfileApiClient::SAFEGUARDING_FLAGS[:owner])
+  end
+
+  it 'does not create the school teacher safeguarding flag when logged in as a teacher' do
+    teacher = create(:teacher, school:)
+    authenticated_in_hydra_as(teacher)
+
+    delete("/api/schools/#{school.id}/students/#{student_id}", headers:)
+    expect(ProfileApiClient).not_to have_received(:create_safeguarding_flag).with(token: UserProfileMock::TOKEN, flag: ProfileApiClient::SAFEGUARDING_FLAGS[:teacher])
+  end
```

Comment:
❓ should the teacher safeguarding flags not be set in this case?

Reply:
Not in this case. We have [a rule in our ability class that prevents teachers deleting students](https://github.com/RaspberryPiFoundation/editor-api/blob/f82f10f7d323e151fbab4a7b04d8ee9890cb34af/app/models/ability.rb#L70) which means that teachers are rejected by the [call to `authorize_resource :student` in the SchoolStudentsController](https://github.com/RaspberryPiFoundation/editor-api/blob/f82f10f7d323e151fbab4a7b04d8ee9890cb34af/app/controllers/api/school_students_controller.rb#L7).

Reply:
aha of course 😄 thanks